src/Access/ContextAccess.cpp

```diff
@@ -168,6 +168,12 @@ AccessRights ContextAccess::addImplicitAccessRights(const AccessRights & access,
             res |= show_databases;
         }
 
+        static const AccessFlags alter_delete = AccessType::ALTER_DELETE;
+        static const AccessFlags select = AccessType::SELECT;
+        static const AccessFlags move_partition = AccessType::ALTER_MOVE_PARTITION;
+        if ((res & alter_delete) && (res & select) && level <= 2)
```

Comment:
Is the new `level <= 2` requirement backward compatible?

Reply:
No, the `ALTER_MOVE_PARTITION` grant can be granted only on a database (level=1) or table (level=2). It can't be granted on a column